src/index.js

```diff
@@ -41,15 +42,28 @@ export default function ({ Plugin, types: t }) {
     return cls.body.body.filter(isRenderMethod).length > 0;
   }
 
-  const isCreateClassCallExpression = t.buildMatchMemberExpression('React.createClass');
+  function isCreateClassCallExpression(node, factoryMethods) {
+    for (const method of factoryMethods) {
+      if (method.indexOf('.') !== -1) {
+        if (t.buildMatchMemberExpression(method)(node.callee)) {
```

Comment:
This appears to have worse performance. (I haven't tested but the code looks so.) The point of `buildMatchMemberExpression` is to *build the matching function and save it*. However in this code we build the matching function on every invocation.

I think what we really need here is `buildMatchCreateClassCallExpression(factoryMethods)` which would *return* a `node`-accepting function. This way we should be able to create (and memoize) the matching function once per plugin invocation. Does this make sense?